### ISSUE TITLE
Using a nib

### ISSUE THREAD

**Issue by Luflosi** [kovidgoyal/kitty#1625]
From `glow/cocoa_window.m`:
```C
// Set up the menu bar (manually)
// This is nasty, nasty stuff -- calls to undocumented semi-private APIs that
// could go away at any moment, lots of stuff that really should be
// localize(d|able), etc.  Add a nib to save us this horror.
```
Do you know how to create and use this nib file or where to get that information? I didn't find any documentation in the glfw docs. I also didn't really find any useful information anywhere else. The nib file seems to be related to a xib file.

**Comment by kovidgoyal** [comment]
There is no need. kitty creates the menubar programmatically see
kitty/cocoa_window.m. IIRC the code from glfw is not used at all.

**Comment by Luflosi** [comment]
OK, so we don't use that glfw code but a slightly modified piece of code. The comment is still true, we do use semi-private APIs and it's not localizable. I'd like to know where to get more information about creating a nib file.

**Comment by kovidgoyal** [comment]
None of kitty is localized. So I dont see what the point of starting with the macOS specific menubar is. And I have no idea where to find out more about nibs.

**Comment by Luflosi** [comment]
I had very limited success.
Here is what I did and what didn't work for anyone else who might be interested:
I used Xcode to create a new Cocoa App named kitty, with the storyboard option disabled. I then renamed everything called `AppDelegate` to `GlobalMenuTarget`. I then clicked on MainMenu.xib to open the interface builder or whatever it's called. I then opened the kitty menu bar item, where there is a `Preferences…` entry, right clicked onto that item and dragged from the little plus from the action to a good place in `GlobalMenuTarget.m`. I named the function `show_preferences` because that's how it's called in kitty. This should execute that function when trying to open the preferences later. This works correctly when starting this app. I then started the app with the triangle and following that opened the app in finder by right clicking on the `kitty.app` entry  under `Products`. I opened the `kitty.app` contents and copied the director(y/ies) in the Resources directory into the app bundle of the real kitty. To make kitty actually load the file(s) I had to slightly modify the code by putting
```Objective-C
    if ([[NSBundle mainBundle] pathForResource:@"Base.lproj/MainMenu" ofType:@"nib"])
    {
        static id                  nibObjects;
        printf("loading MainMenu\n");
        [[NSBundle mainBundle] loadNibNamed:@"Base.lproj/MainMenu"
                                      owner:NSApp
                            topLevelObjects:&nibObjects];
    }
    else {
        printf("cocoa_create_global_menu\n");
```
into `cocoa_create_global_menu()`.
Unfortunately, after starting kitty, the preferences menu item is greyed out, probably because the nib can't find the appropriate code to call. I was unable to fix that, even after a couple hours of trying different things. There were no error messages related to the menu bar on stdout or in Console.app as far as I can tell.
I'll give up on this idea for now until I learn something new that might help me.
If anyone has any suggestions, I would very much appreciate them.
